### PR TITLE
Update dependency on `guidelines-checker` to newly republished name

### PR DESCRIPTION
**Pull request by scbedd**
Also ensure that we clean up old code.

@l0lawrence for review and approval.

### CODE REVIEW

File: eng/apiview_reqs.txt [hunk 1]
```diff
@@ -1,2 +1,2 @@
 api-stub-generator==0.3.5
-pylint-guidelines-checker==0.0.7
+azure-pylint-guidelines-checker==0.0.7
```
Comment (scbedd) [inline eng/apiview_reqs.txt:2]: Need to publish 0.0.7.

Have published:
- [x] `0.0.8`
- [x] `0.0.5`

**Review by l0lawrence — COMMENTED** [review]
Does the pylintrc file also need to be updated to load-plugin==azure_pylint_guidelines_checker

**Comment by scbedd** [comment]
> Does the pylintrc file also need to be updated to load-plugin==azure_pylint_guidelines_checker

Great question. I left the module named the same, `pylint_guidelines_checker` to avoid a lot of churn. I _thought_ it would be the same, but maybe I'm wrong.

**Comment by l0lawrence** [comment]
> > Does the pylintrc file also need to be updated to load-plugin==azure_pylint_guidelines_checker
> 
> Great question. I left the module named the same, `pylint_guidelines_checker` to avoid a lot of churn. I _thought_ it would be the same in the rcfile as a result, but maybe I'm wrong. Waiting for build outcome.

Based on locally running it, it looks like it still pulls in the custom with it remaining pylint_guidelines_checker :)

**Review by l0lawrence — APPROVED** [review]

**Pull request merged**